DSC/dsc.py

```diff
@@ -125,16 +129,17 @@ def enable():
     try:
         start_omiservice()
         mode = get_config('Mode')
+        waagent.AddExtensionEvent(name=ExtensionShortName, op='EnableInprogress', isSuccess=True, message="Enabling the DSC extension - mode: " + mode)
         if mode == '':
             mode = Mode.push
         else:
             mode = mode.lower()
             if not hasattr(Mode, mode):
                 waagent.AddExtensionEvent(name=ExtensionShortName,
                                           op=Operation.Enable,
-                                          isSuccess=False,
+                                          isSuccess=True,
                                           message="(03001)Argument error, invalid mode")
-                hutil.do_exit(1, 'Enable', 'error', '1', 'Enable failed, unknown mode: ' + mode)
+                hutil.do_exit(51, 'Enable', 'error', '51', 'Enable failed, unknown mode: ' + mode)
```

Comment:
DSC supports either pull or push mode. if users enters incorrect mode, then it is an argument error and exit code is 51. We don't want to consider this in our score card.